### ISSUE TITLE
用YOLOv3训练时，迭代次数无法根据设定值改变

### ISSUE THREAD

**Issue by celineJiang** [PaddlePaddle/Paddle#16577]
- 版本、环境信息：
    1）PaddlePaddle版本：1.3fluid
    2）系统环境：Ubuntu 16.04、Python 3.5
- 问题描述：
在使用YOLOv3训练自己的数据集时，发现iter的次数只与batch_size挂钩（不确定是否也与数据集大小有关），即：当数据集大小为1913时，若设batch_size为16，则iter打印次数只有116次；若设batch_size为32，则iter打印次数只有57次。而在代码中明确写了：
```python
for iter_id in range(0, 500200):
``````
因此我感到困惑，不知道是哪里出现了问题，希望能给出解释或告诉我可以从哪里入手去解决它，感谢！另附重要代码：
```python
    def train_loop_pyreader():
        py_reader.start()
        smoothed_loss = SmoothedValue()
        try:
            start_time = time.time()
            snapshot_loss = 0
            snapshot_time = 0
            for iter_id in range(0, 500200):
                prev_start_time = start_time
                start_time = time.time()
                losses = exe.run(fetch_list=[v.name for v in fetch_list])
                smoothed_loss.add_value(np.mean(np.array(losses[0])))
                snapshot_loss += np.mean(np.array(losses[0]))
                snapshot_time += start_time - prev_start_time
                lr = np.array(fluid.global_scope().find_var('learning_rate').get_tensor())
                print("Iter {:d}, lr {:.6f}, loss {:.6f}, time {:.5f}".format(
                    iter_id, lr[0],
                    smoothed_loss.get_mean_value(), start_time - prev_start_time))
                sys.stdout.flush()
                if (iter_id + 1) % cfg.snapshot_iter == 0:
                    save_model("model_iter{}".format(iter_id))
                    print("Snapshot {} saved, average loss: {}, average time: {}".format(
                        iter_id + 1, snapshot_loss / float(cfg.snapshot_iter),
                        snapshot_time / float(cfg.snapshot_iter)))
                    snapshot_loss = 0
                    snapshot_time = 0
        except fluid.core.EOFException:
            py_reader.reset()
``````

**Comment by heavengate** [comment]
你实现的自己数据集的reader是不是读完一个epoch就不再读了

**Comment by heavengate** [comment]
https://github.com/PaddlePaddle/models/blob/9cd7d08b6f5806d16b2acb8e796112963a745bcb/fluid/PaddleCV/yolov3/reader.py#L215   YOLOv3的reader读完一个epoch后会shuffle数据重新读，reader没有中止，通过train脚本控制总iter数

**Comment by celineJiang** [comment]
> https://github.com/PaddlePaddle/models/blob/9cd7d08b6f5806d16b2acb8e796112963a745bcb/fluid/PaddleCV/yolov3/reader.py#L215 YOLOv3的reader读完一个epoch后会shuffle数据重新读，reader没有中止，通过train脚本控制总iter数

您好，问题已经得到解决了，谢谢您的回答！学习了！
